### PR TITLE
#9468 Backport Circle CI configuration on 2.3

### PR DESCRIPTION
**Pull request by arnolanglade**
**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+          - run:
+                name: Start containers
+                command: docker-compose up -d fpm
+          - run:
+                name: PhpSpec
+                command: docker-compose exec -T fpm vendor/bin/phpspec run --format=junit > var/tests/phpspec/specs.xml
+          - run:
+                name: PhpCsFixer
+                command: docker-compose exec -T fpm vendor/bin/php-cs-fixer fix --diff --dry-run --config=.php_cs.php
+          - run:
```
Comment (ahocquard) [inline .circleci/config.yml]: missing coupling detector

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+                command: docker-compose up -d
+          - run:
+                name: Install database
+                command: |
+                    sleep 25 # wait mysql to be up
+                    docker-compose exec fpm bin/console --env=behat pim:install:db
+          - run:
+                name: PHPunit End to end
+                command: |
+                    TESTFILES=$(circleci tests glob "src/**/*EndToEnd.php" | circleci tests split)
```
Comment (grena) [inline .circleci/config.yml]: I don't find any file on CE 2.3 matching this pattern

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+workflows:
+    version: 2
+    pull_request:
+        jobs:
+            - wait_for_user_approval:
+                  type: approval
+                  filters:
+                      branches:
+                          ignore:
+                              - master
```
Comment (grena) [inline .circleci/config.yml]: To remove

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+                name: PHPunit Integration
+                command: |
+                    TESTFILES=$(circleci tests glob "tests/back/**/*Integration.php" "src/**/*Integration.php" | circleci tests split)
+                    .circleci/run_phpunit.sh $TESTFILES
+          - store_test_results:
+                path: var/tests/phpunit
+          - store_artifacts:
+                path: var/tests/phpunit
+
+  back_phpunit_end_to_end:
```
Comment (ahocquard) [inline .circleci/config.yml]: to remove as EndToEnd is the name of the directory

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+                  requires:
+                      - back_phpunit_integration
+            - back_behat_legacy:
+                  requires:
+                      - back_phpunit_end_to_end
+
+    nightly:
+        triggers:
+            - schedule:
+                  cron: "0 2 * * *"
```
Comment (ahocquard) [inline .circleci/config.yml]: 3

File: .circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+                  requires:
+                      - setup
+            - front_static_acceptance_and_integration:
+                  requires:
+                      - setup
+            - back_phpunit_integration:
+                  requires:
+                      - back_static_and_acceptance
+                      - front_static_acceptance_and_integration
+            - back_phpunit_end_to_end:
```
Comment (ahocquard) [inline .circleci/config.yml]: to remove

File: .circleci/config.yml (hunk 1 above)
Comment (BitOne) [reply .circleci/config.yml]: added

File: .circleci/config.yml (hunk 2 above)
Comment (BitOne) [reply .circleci/config.yml]: Removed

File: .circleci/config.yml (hunk 1 above)
Comment (BitOne) [reply .circleci/config.yml]: Added

File: .circleci/config.yml (hunk 4 above)
Comment (BitOne) [reply .circleci/config.yml]: Done

File: .circleci/config.yml (hunk 6 above)
Comment (BitOne) [reply .circleci/config.yml]: Done

File: .circleci/config.yml (hunk 5 above)
Comment (BitOne) [reply .circleci/config.yml]: done

File: .circleci/config.yml [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+        - run:
+            name: Front acceptance tests
+            command: MAX_RANDOM_LATENCY_MS=100 docker-compose run --rm node yarn run acceptance ./tests/features
+workflows:
+    version: 2
+    pull_request:
+        jobs:
+            - wait_for_user_approval:
+                  type: approval
+                  filters:
```
[the remark this answers is not in the document]
Comment (BitOne) [reply .circleci/config.yml]: Nope :p

[the remark this answers is not in the document]
Comment (BitOne) [reply .circleci/config.yml]: Nope :p

**Pull request merged**